aws/table_aws_route53_resolver_endpoint.go

```diff
@@ -0,0 +1,318 @@
+				Transform:   transform.FromField("Tags").Transform(route53resolverTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsRoute53ResolverEndpoint(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAwsRoute53ResolverEndpoint", "AWS_REGION", region)
+
+	// Create session
+	svc, err := Route53ResolverService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.ListResolverEndpointsPages(
+		&route53resolver.ListResolverEndpointsInput{},
+		func(page *route53resolver.ListResolverEndpointsOutput, isLast bool) bool {
+			for _, parameter := range page.ResolverEndpoints {
+				d.StreamListItem(ctx, parameter)
+
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsRoute53ResolverEndpoint(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsSSMParameter")
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// Create Session
+	svc, err := Route53ResolverService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.GetResolverEndpointInput{
+		ResolverEndpointId: &id,
+	}
+
+	// Get call
+	data, err := svc.GetResolverEndpoint(params)
+	if err != nil {
+		logger.Debug("getAwsRoute53ResolverEndpoint", "ERROR", err)
+		return nil, err
+	}
+
+	return data.ResolverEndpoint, nil
+}
+
+func listResolverEndpointIPAddresses(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listResolverEndpointIpAddresses")
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	resolverEndpointData := h.Item.(*route53resolver.ResolverEndpoint)
+
+	// Create Session
+	svc, err := Route53ResolverService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.ListResolverEndpointIpAddressesInput{
+		ResolverEndpointId: resolverEndpointData.Id,
+	}
+
+	op, err := svc.ListResolverEndpointIpAddresses(params)
+	if err != nil {
+		logger.Debug("listResolverEndpointIpAddresses", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func listResolverRule(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
Since we have rules as a separate table, and there's no rule info in the native describe/list calls for endpoints, we should not add this information here. Users can instead use joins to link endpoint and rule information together.

Reply:
done with changes.